src/java/org/apache/cassandra/db/lifecycle/LifecycleTransaction.java

```diff
@@ -104,7 +104,7 @@ public String toString()
     private final Set<UniqueIdentifier> identities = Collections.newSetFromMap(new IdentityHashMap<UniqueIdentifier, Boolean>());
 
     // changes that have been made visible
-    private final State logged = new State();
+    public final State logged = new State();
```

Comment:
nit: can you just make a method that returns an immutablecopy of logged.obsolete instead of changing access here